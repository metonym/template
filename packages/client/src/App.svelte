<script lang="ts">
import Query from "@components/Query.svelte";
import { QueryClient, QueryClientProvider } from "@tanstack/svelte-query";

const queryClient = new QueryClient();
</script>

<QueryClientProvider client={queryClient}>
  <Query />
</QueryClientProvider>
