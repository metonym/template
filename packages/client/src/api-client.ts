import { createTRPCProxyClient, httpBatchLink } from "@trpc/client";
import type { Router } from "server/src/trpc-router";

export const trpcClient = createTRPCProxyClient<Router>({
  links: [httpBatchLink({ url: "/api" })],
});
