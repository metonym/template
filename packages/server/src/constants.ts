export const PORT = import.meta.env.PORT ?? 3000;
export const SITE_URL = import.meta.env.SITE_URL ?? `http://localhost:${PORT}/`;
export const NODE_ENV = import.meta.env.NODE_ENV ?? "development";
