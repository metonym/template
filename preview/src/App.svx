# {name} v{version}

<script>
  import Button from "component-svelte";

  let version = process.env.VERSION;
  let name = process.env.NAME;
  let count = 10;
</script>

<style>
  :global(*) {
    font-family: sans-serif;
  }
</style>

## {count}

<Button bind:count />
