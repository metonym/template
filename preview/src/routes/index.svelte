<script>
  import Component from "component-svelte";
</script>

<Component />
