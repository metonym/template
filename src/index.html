<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <script>
      var VERSION = "<%= VERSION %>";
    </script>
  </head>
  <body></body>
</html>
