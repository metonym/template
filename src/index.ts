import { parse, walk } from "svelte/compiler";
import MagicString from "magic-string";

interface Options {
  ignore?: boolean;
}

export const preprocessor: SveltePreprocessor<"markup", Options> = (
  options
) => {
  const ignore = options?.ignore === true;

  return {
    markup({ content, filename }) {
      if (ignore) return;
      if (filename && /node_modules/.test(filename)) return;

      const ast = parse(content);
      const s = new MagicString(content);
      const class_names = new Set();

      walk(ast, {
        enter(node: AstNode, parent: AstNode) {
          if (node.type === "InlineComponent") {
            const class_attribute = node.attributes.find(
              ({ type, name }: any) => type === "Attribute" && name === "class"
            );

            class_attribute?.value[0]?.raw
              .split(" ")
              .forEach((class_name: string) => {
                class_names.add(class_name);
              });
          }

          if (node.type === "ClassSelector" && class_names.has(node.name)) {
            if ((parent as SelectorNode).children.length > 1) {
              const selector = content.slice(parent.start, parent.end);
              s.overwrite(parent.start, parent.end, `:global(${selector})`);
            } else {
              s.overwrite(node.start, node.end, `:global(.${node.name})`);
            }
          }
        },
      });

      return {
        code: s.toString(),
        map: s.generateMap({
          source: filename,
          hires: true,
        }),
      };
    },
  };
};
