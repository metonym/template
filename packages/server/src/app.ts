import { trpcServer } from "@hono/trpc-server";
import { Hono } from "hono";
import { serveStatic } from "hono/bun";
import { csrf } from "hono/csrf";
import { logger } from "hono/logger";
import { secureHeaders } from "hono/secure-headers";
import { NODE_ENV, PORT, SITE_URL } from "./constants";
import { router } from "./trpc-router";

const app = new Hono();

if (NODE_ENV === "production") {
  app
    .use(secureHeaders())
    .use(csrf({ origin: new URL(SITE_URL).host }))
    .use(logger())
    .use("/*", serveStatic({ root: "./dist" }));
  console.log(NODE_ENV, SITE_URL);
}

app.use("/api/*", trpcServer({ router }));

export default {
  port: PORT,
  fetch: app.fetch,
};
