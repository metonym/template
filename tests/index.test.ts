import type { Processed } from "svelte/types/compiler/preprocess";
import { expect, test } from "vitest";
import * as API from "../src";

test("API", () => {
  expect(Object.keys(API)).toMatchInlineSnapshot(`
    [
      "preprocessor",
    ]
  `);
});

const preprocess = (content: string) =>
  (API.preprocessor().markup?.({ content }) as Processed).code;

test("no styles", () => {
  expect(
    preprocess(`<Component id="my-id" class="class1 class2" />`)
  ).toMatchSnapshot();
});
