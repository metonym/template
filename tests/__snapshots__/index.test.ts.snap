// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no styles 1`] = `"<Component id=\\"my-id\\" class=\\"class1 class2\\" />"`;
